charts/graph-network-indexer/templates/indexer-agent/statefulset.yaml

```diff
@@ -51,42 +52,51 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
-          {{- with (deepCopy $.Values.indexerDefaults.env | mustMerge $values.env) }}
-          {{- range $key, $val := .}}
-          {{- if not (kindIs "invalid" $val) }}
+          {{- if $values.env }}
+          {{- range $key, $val := $values.env }}
             - name: {{ $key | quote }}
               value: {{ $val | quote }}
           {{- end }}
           {{- end }}
-          {{- end }}
-          {{- with (deepCopy $.Values.indexerDefaults.secretEnv | mustMerge $values.secretEnv) }}
-          {{- range $key, $val := .}}
+          {{- if $values.secretEnv }}
+          {{- range $key, $val := $values.secretEnv }}
             - name: {{ $key | quote }}
               valueFrom:
                 secretKeyRef:
                   name: {{ $val.secretName | quote }}
                   key: {{ $val.key | quote }}
-                  optional: false
           {{- end }}
           {{- end }}
           args:
-          {{- with (deepCopy $.Values.indexerDefaults.config | mustMerge $values.config) }}
-          {{- range $key, $val := .}}
-          {{- if not (kindIs "invalid" $val) }}
-            - --{{ $key }}={{ $val }}
-          {{- end }}
-          {{- end }}
-          {{- end }}
+            - --graph-node-query-endpoint={{ index $values.config.graph_node "query_url" }}
```

Comment:
the dot notion can be used with keys with underscores, as $values.config.graph_node shows. Why are we going with "index" in a generalized way?